# High Point Sofa and Chair

In [16]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

# Importe de Datos
data = pd.read_csv("./HighPointSofaandChair.csv")
data.head()

Worker  Performance  Aptitude  Union
0    Abbott           58         5      0
1  Anderson           53         4      0
2    Bender           33        10      0
3      Bush           97        10      0
4    Center           36         2      0

# 1. Determinar las variables independientes

In [17]:
X = data[['Aptitude', 'Union']]
X 

Aptitude  Union
0          5      0
1          4      0
2         10      0
3         10      0
4          2      0
5          7      0
6          6      0
7          9      0
8          9      1
9          2      1
10         8      1
11         6      1
12         7      1
13         3      1
14         6      1

# #2. Determinar la variable dependiente

In [18]:
y = data[['Performance']]
y

Performance
0            58
1            53
2            33
3            97
4            36
5            83
6            67
7            84
8            98
9            45
10           97
11           90
12           96
13           66
14           82

In [19]:
# Ajustar el modelo de regresión múltiple
model = LinearRegression()
model.fit(X, y) #Entrenando el modelo
y_pred = model.predict(X) #Predicciones del modelo
y_pred

#Métricas estadísticas
mse = mean_squared_error(y, y_pred) # Error de la media cuadrática
r2 = r2_score(y, y_pred) # Varianza

# 3. Valor de Intersección

In [20]:
model.intercept_

array([29.27618014])

In [21]:
# Coeficientes de la regresión
print("Coeficientes: ", model.coef_)
print("Error de la media cuadrática (MSE): ", mse)
print("Varianza (R^2): ", r2)

Coeficientes:  [[ 5.22246337 22.13510581]]
Error de la media cuadrática (MSE):  228.93615482003986
Varianza (R^2):  0.5329951510923938


# 4. Estimación del Rendimiento del Trabajador

In [22]:
Y = model.predict(np.array([[7,1]]))
print("Rendimiento estimado del trabajador: ", Y)

Rendimiento estimado del trabajador:  [[87.96852957]]


c:\Users\panch\anaconda3\Lib\site-packages\sklearn\utils\validation.py:2749: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


## Conclusión
En este ejercicio analizamos qué factores explican mejor el desempeño laboral de los electricistas.
Usamos dos variables:
- la calificación del examen de aptitud, y si el trabajador pertenece o no a un sindicato.

El modelo de regresión muestra que sí existe una relación entre estas variables y el desempeño.
El R² es aproximadamente 0.53, lo que significa que alrededor del 53% de la variación del desempeño laboral puede explicarse con estas dos variables.